### ISSUE TITLE
Alert dismiss button is missing accessible label

### ISSUE THREAD

**Issue by hbuchel** [aws-amplify/amplify-ui#2125]
### Before creating a new issue, please confirm:

- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-ui/issues?q=is%3Aissue+) and [discussions](https://github.com/aws-amplify/amplify-ui/discussions).
- [X] I have read the guide for [submitting bug reports](https://github.com/aws-amplify/amplify-ui/blob/main/CONTRIBUTING.md#bug-reports).

### On which framework/platform are you having an issue?

React

### Which UI component?

Other

### How is your app built?

Any

### Please describe your bug.

The Alert primitive is missing an accessible label for the dismiss button.

```
{isDismissible && (
  <Button
    variation="link"
    className={ComponentClassNames.AlertDismiss}
    onClick={dismissAlert}
    ref={buttonRef}
  >
    <IconClose />
  </Button>
)}
```

### What's the expected behaviour?

Dismiss button should have a discernible name. We should use either an aria-label for the button or add VisuallyHidden text and make that text customizable via a prop.

### Help us reproduce the bug!

https://ui.docs.amplify.aws/react/components/alert#dismissible

### Code Snippet

```typescript
// Put your code below this line.

```

### Additional information and screenshots

<img width="1100" alt="Screen Shot 2022-06-16 at 12 16 04 PM" src="https://user-images.githubusercontent.com/376920/174118037-ab31361c-f163-40ca-b356-7ddac206af78.png">

**Comment by hbuchel** [comment]
FYI, this issue also exists for Vue in [base.alert](https://github.com/aws-amplify/amplify-ui/blob/main/packages/vue/src/components/primitives/base-alert.vue#L33) and for Angular in [error.component](https://github.com/aws-amplify/amplify-ui/blob/main/packages/angular/projects/ui-angular/src/lib/primitives/error/error.component.html) .
